### ISSUE TITLE
[1.13 UX] docker service ps -a

### ISSUE THREAD

**Issue by aaronlehmann** [docker/docker#28507]
As of #25983, `docker service ps` now needs the `-a` option to show tasks that failed or completed. This seems okay in cases where the service has running tasks and we simply don't show the historic tasks, but it seems confusing when nothing is running and you're trying to figure out why. It's a big UX change from 1.12 to need to run `docker service ps -a` in this situation.

As a compromise, I would suggest showing completed/failed tasks by default only if there is no running task in that slot. With the present (new) behavior, information about those slots doesn't show up at all unless you use `-a`.

cc @aluzzardi @thaJeztah

**Comment by jhorwit2** [comment]
This is how `docker ps` works (it won't show anything if none are running), so should that be changed also to be consistent?

This change would require the CLI to implement filtering logic instead of using the filter logic already present in the swarm manager/store.

**Comment by aaronlehmann** [comment]
I understand the consistency argument, but on the other hand, changing the behavior from one version to another can be surprising, even if it's in the name of consistency.

**Comment by aluzzardi** [comment]
That's what we originally had (`-a`) and then we changed it before 1.12 shipped in order to display all tasks and do naming indentation in order to get a better history.

This change is going backwards.

The difference with `docker ps` is:
- Tasks are garbage collected - there's going to be (by default) 5 by replica. docker ps just piles them up
- Parent tasks are related. Those are the ancestors of this task, not just random tasks
- Tasks are one shot. If a task crashes, it's easy to see it crashed and when. 

The latter is the reason we made this change in the first place in 1.12 - users were confused as it wasn't obvious that a task was crashing.

@samalba just hit the same problem earlier today when trying 1.13
